### PR TITLE
Only update outdated_parsers on TSUpdate

### PR DESCRIPTION
**Pull request by theHamsta**
I have no idea why Github decided to close #640 but here is an updated version. It should now also work when `lockfile.json` changed after a plugin update and when `parser-info` is not writable.

Please also test. E.g. by

-  `:TSUpdate` (now it should write all installed parser revisions, will run for all parsers a last time)
- Change `lockfile.json` (only one revision -> should only update that parser on TSUpdate, or execute `write-lockfile.json` -> :TSUpdate should  update all parsers with new revision available)

### CODE REVIEW

**Comment by theHamsta** [comment]
Bonus: fixes README.md CI

File: README.md [hunk 1]
```diff
@@ -226,7 +226,6 @@ List of currently supported languages:
 
 <!--This section of the README is automatically updated by a CI job-->
 <!--parserinfo-->
-
```
Comment (theHamsta) [inline README.md:229]: Ok, this was the reason? Someone changed that section manually while it still was the same for the Lua script?

Comment (theHamsta) [reply README.md:229]: Exeriment. Will CI fail when this is fixed manually?

Comment (theHamsta) [reply README.md:229]: Result: it will work when REAMDE needs update and also if not.

**Review by vigoux — APPROVED** [review]

**Pull request merged**